### DISCUSSION TITLE
get values of enum

### DISCUSSION THREAD

**Discussion by sjohns21** [supabase/supabase#18536] [Feature Requests]
i have a native_language enum
i'd like to have direct access to these values in my client code (typescript)
but the client type output is a string union
and i don't see how to turn this into an array of values

**Comment by infinito84** [comment]
I use this:

```
CREATE OR REPLACE FUNCTION get_types (enum_type text) 
RETURNS json AS $$
DECLARE
  json_data json;
  text_query text;
BEGIN 
  text_query := format (
    'SELECT array_to_json(enum_range(NULL::%s))',
    enum_type
  );

  EXECUTE text_query INTO json_data;

  RETURN json_data;
END 
$$ LANGUAGE 'plpgsql';

```

Then I call `/rest/v1/rpc/get_types?enum_type=SomeEnum`

or

```
let { data } = await supabase.rpc('get_types', { enum_type: 'SomeEnum'  })
```

**Reply by Yorktech** [reply]
You can create a view for all your lookups like so

create view lookups as 

SELECT 'enum1' as enum_type, unnest(enum_range(NULL::enum1) ::VARCHAR as enums  union
SELECT  'enum2' as enum_type, unnest(enum_range(NULL::enum2)) ::VARCHAR as enums union
SELECT  'enum3' as enum_type, unnest(enum_range(NULL::enum3)) ::VARCHAR as enums union
SELECT  'enum4' as enum_type, unnest(enum_range(NULL::enum4)) ::VARCHAR as enums	

Javascript

const { data: lookups, error3 } = await supabase
      .from('lookups')
      .select();
      
 And filter the lookups say for a dropdown list
const dropdownlist = .lookups.filter(item => item.name === 'enum1');
